tests/src/rules/order.js

```diff
@@ -1153,7 +1153,91 @@ ruleTester.run('order', rule, {
         },
       ],
     }),
+    // reorder fix cannot cross function call on moving below #1
+    test({
+      code: `
+        const local = require('./local');
+
+        fn_call();
+
+        const global1 = require('global1');
+        const global2 = require('global2');
+
+        fn_call();
+      `,
+      output: `
+        const local = require('./local');
+
+        fn_call();
+
+        const global1 = require('global1');
+        const global2 = require('global2');
+
+        fn_call();
+      `,
+      errors: [{
+        ruleId: 'order',
+        message: '`./local` import should occur after import of `global2`',
+      }],
+    }),
+    // reorder fix cannot cross function call on moving below #2
+    test({
+      code: `
+        const local = require('./local');
+        fn_call();
+        const global1 = require('global1');
+        const global2 = require('global2');
 
+        fn_call();
+      `,
+      output: `
+        const local = require('./local');
+        fn_call();
+        const global1 = require('global1');
+        const global2 = require('global2');
+
+        fn_call();
+      `,
+      errors: [{
+        ruleId: 'order',
+        message: '`./local` import should occur after import of `global2`',
+      }],
+    }),
+    // reorder fix cannot cross function call on moving below #3
+    test({
+      code: `
+        const local1 = require('./local1');
+        const local2 = require('./local2');
+        const local3 = require('./local3');
```

Comment:
All of these new tests start with each group already ordered; can we unordered them, and show that the autofixer properly orders them?

Reply:
@ljharb Great! I wrote requested tests and found some cases that autofixer dont order imports inside group. I inspected code and find out that ability to reorder imports inside group requires great rework of rule code (another issue?). 

This PR just fix a bug in current algorithm of rule. I suggest use this PR as quick fix, and create another (less critical) issue to fix sorting inside groups.

Reply:
In that case I'd still like to see the test cases show what it *currently* does when it starts out out of order; perhaps with additional cases instead of changing these?

Reply:
Added few tests with unordered groups around function call, including cases with not fully reordered groups after applying autofix